uimaj-core/src/main/java/org/apache/uima/pear/util/XMLUtil.java

```diff
@@ -828,4 +862,18 @@ public static void printXMLTag(String tag, Properties attributes, PrintWriter oW
     oWriter.print('>');
     oWriter.flush();
   }
+  
+  private static String xmlEscape(String value)
+  {
+      if (value == null) {
+          return value;
+      }
+      
+      return value
+              .replace("&", "&amp;")
```

Comment:
I wonder if the user has already written &lt; for example, would this result in &amplt; ? In other words, does the replacement need  to first confirm that the character being replaced, isn't already in the right escape format?

Reply:
I expect if the user had already written `&amp;`, then it is because the path includes `&amp;` and not because the user wants the the stuff being unescaped. 

I mean if you want to install a PEAR in `/tmp/&amp;foldername`, then you don't want to loo for it in `/tmp/&foldername`, right?

Reply:
Btw. I was considering to use XML escaping methods form commons-lang3 or commons-text, but didn't want to introduce the dependency. I was also considering to port this whole stuff over to Xmlizable (i.e what other UIMA descriptors use) or even JAXB - but again, it seemed to much for the time being.

Reply:
This apache commons project has xml escape code for both xml version 1.0 and 1.1.  It seems it might do more than this.  I think it would be worth a look...

Reply:
https://commons.apache.org/proper/commons-text/javadocs/api-release/org/apache/commons/text/StringEscapeUtils.html

Reply:
So, your point is that this xml cleanup code is only used within the Pear context, where it would not be reasonable for anyone to have inputted the & char in an attempt to escape something?

Reply:
Yep. Also, the thing that in particular the path is not only substituted in the XML file but also used to actually control the location where the pear is installed. So if it contained an `&amp;`, then the folder name would include the `&amp;` but when parsing the XML descriptor the `&amp;` would become a `&` and would no longer match the installation path.

Reply:
Incidentally, when I first heard of the issue, I even suggested to simply escape the path before passing it to the PEAR installer, but got told it had already been tried and also hadn't worked.